R/mod-upload-documentation.R

```diff
@@ -202,3 +140,106 @@ upload_documents_server <- function(input, output, session,
     }
   })
 }
+
+#' Create instructions for uploading documentation
+#'
+#' This function creates the instruction text for the Documentation
+#' portion of dccvalidator. The study_link_ref argument dictates
+#' whether instructions for uploading an acknowledgment is added.
+#' If study_link_ref is an empty string, then the acknowledgement
+#' instructions are not added.
+#'
+#' @inheritParams upload_documents_ui
+#' @return html taglist with instructions
+upload_docs_instruct_text <- function(study_link_human, study_link_animal,
+                                      study_link_ref) {
+  if (study_link_ref != "") {
+    overview_text <- p(
+      # nolint start
+      "This information should be similar to a materials and methods section in a paper. An example of what a study should include can be found ",
+      HTML(glue::glue("<a target =\"_blank\" href=\"{study_link_animal}\">here</a> for an animal model study and ")),
+      HTML(glue::glue("<a target =\"_blank\" href=\"{study_link_human}\">here</a> for a human study.")),
+      "If you wish, also provide an acknowledgement statment and/or reference that should be included in publications resulting from secondary data use; examples can be found ",
+      HTML(glue::glue("<a target =\"_blank\" href=\"{study_link_ref}\">here</a>.")),
+      "This can be provided as part of the study or assay documentation text."
```

Comment:
To simplify things I suggest we tell people to provide it with the study description:

```suggestion
      "This can be provided as part of the study documentation text."
```

Reply:
Make consistent with instructions.